### PR TITLE
feat: custom rpc assets

### PR DESCRIPTION
**Pull request by estebanmino**
Fixes BX-####
Figma link (if any):

## What changed (plus any additional context for devs)

- custom rpc groups in settings
- custom rpc assets in settings
- added `useCustomNetwork` to use it in the same way we use wagmi `useNetwork` to get custom rpcs 
- added `getCustomChains` so we can call it from createWagmiClient in the same way we have wagmi's `getChains`

## Screen recordings / screenshots

https://www.loom.com/share/9c849c0ad10c4897a4bb144a5fc98ecc

### CODE REVIEW

File: src/core/state/customRPCAssets/index.ts [hunk 1]
```diff
@@ -0,0 +1,93 @@
+import { Address } from 'wagmi';
+import create from 'zustand';
+
+import { createStore } from '../internal/createStore';
+
+export interface CustomRPCAsset {
+  address: Address;
+  decimals: number;
+  symbol: string;
```
Comment (brunobar79) [inline src/core/state/customRPCAssets/index.ts:9]: I think it should have a name too

File: src/entries/popup/pages/settings/customRPC/customRPC.tsx [hunk 2]
```diff
@@ -0,0 +1,249 @@
...
+            <Input
+              onChange={(t) =>
+                onInputChange<string>(t.target.value, 'string', 'address')
+              }
+              height="32px"
+              placeholder="Address"
+              variant="surface"
+              value={asset.address}
+              // onBlur={onRpcUrlBlur}
+              // borderColor={validations.rpcUrl ? 'accent' : 'red'}
```
Comment (brunobar79) [inline src/entries/popup/pages/settings/customRPC/customRPC.tsx:209]: I assume this code is commented bc this a temporary UI right?

**Review by brunobar79 — APPROVED** [review]
💯

File: src/entries/popup/pages/settings/customRPC/customRPC.tsx (hunk 2 above)
Comment (estebanmino) [reply src/entries/popup/pages/settings/customRPC/customRPC.tsx:209]: will remove

File: src/entries/popup/pages/settings/customRPC/customRPC.tsx [hunk 3]
```diff
@@ -0,0 +1,253 @@
...
+            <Inline alignHorizontal="right">
+              <Button
+                onClick={() =>
+                  removeCustomRPCAsset({ address: asset.address, chainId })
+                }
+                color="accent"
+                height="24px"
+                variant="raised"
+              >
+                Remove Asset
```
Comment (derHowie) [inline src/entries/popup/pages/settings/customRPC/customRPC.tsx:187]: i18n

File: src/entries/popup/pages/settings/customRPC/customRPC.tsx [hunk 4]
```diff
@@ -0,0 +1,253 @@
...
+                  <Inline alignHorizontal="right">
+                    <Button
+                      onClick={() =>
+                        removeCustomRPC({ rpcUrl: customRPC.rpcUrl })
+                      }
+                      color="accent"
+                      height="36px"
+                      variant="raised"
+                    >
+                      Remove
```
Comment (derHowie) [inline src/entries/popup/pages/settings/customRPC/customRPC.tsx:134]: i18n

File: src/entries/popup/pages/settings/customRPC/customRPC.tsx [hunk 5]
```diff
@@ -0,0 +1,253 @@
...
+            />
+
+            <Inline alignHorizontal="right">
+              <Button
+                onClick={addAsset}
+                color="accent"
+                height="36px"
+                variant="raised"
+              >
+                Add
```
Comment (derHowie) [inline src/entries/popup/pages/settings/customRPC/customRPC.tsx:245]: i18n

File: src/entries/popup/pages/settings/customRPC/index.tsx [hunk 6]
```diff
@@ -142,22 +145,46 @@ export function SettingsNetworksCustomRPC() {
...
+                {customChains[Number(chainId)]?.rpcs?.map((customRPC, j) => (
+                  <Box key={j}>
+                    <Inline alignHorizontal="justify">
+                      <Text size="14pt" weight="bold" align="center">
+                        {customRPC.rpcUrl}
+                      </Text>
+                      <Text size="14pt" weight="bold" align="center">
+                        {customRPC.rpcUrl ===
+                        customChains[Number(chainId)].activeRpcUrl
+                          ? 'Active'
```
Comment (derHowie) [inline src/entries/popup/pages/settings/customRPC/index.tsx:178]: i18n

File: src/entries/popup/pages/settings/customRPC/index.tsx [hunk 7]
```diff
@@ -214,6 +249,29 @@ export function SettingsNetworksCustomRPC() {
...
             />
+            <Box padding="10px">
+              <Inline alignHorizontal="justify">
+                <Text
+                  align="center"
+                  weight="semibold"
+                  size="12pt"
+                  color="labelSecondary"
+                >
+                  {'Active'}
```
Comment (derHowie) [inline src/entries/popup/pages/settings/customRPC/index.tsx:260]: i18n

**Review by derHowie — APPROVED** [review]
lgtm, just internationalization nits

File: src/entries/popup/pages/settings/customRPC/index.tsx (hunk 7 above)
Comment (estebanmino) [reply src/entries/popup/pages/settings/customRPC/index.tsx:260]: this is not final ui so i won't add it to languages

**Pull request merged**